### PR TITLE
Added function to upload file in s3 bucket

### PR DESCRIPTION
**Pull request by rakhimundhada15**
Upload the file in s3 bucket.

### CODE REVIEW

File: al_aws.js [hunk 1]
```diff
@@ -216,6 +216,21 @@ var setEnv = function(vars, callback) {
     });
 };
 
+var uploadS3Object = function ({ data, key, bucketName }, callback) {
+    var s3 = new AWS.S3();
+    // Setting up S3 upload parameters
+    const parseData = typeof data !== 'string' ? JSON.stringify(data) : data;
+    const bucket = bucketName ? bucketName : process.env.aws_lambda_s3_bucket
```
Comment (kkuzmin) [inline al_aws.js:223]: let's return an error if `bucketName` is not set.
`aws_lambda_s3_bucket ` is for Lambda packages only

**Review by kkuzmin — CHANGES_REQUESTED** [review]
Please make sure collectors have enough permissions to write into S3

**Review by imranalisyed506 — APPROVED** [review]
LGTM

File: al_aws.js [hunk 2]
```diff
@@ -216,6 +216,25 @@ var setEnv = function(vars, callback) {
...
+            Bucket: bucket,
+            Key: key,
+            Body: parseData
+        };
+        // Uploading files to the bucket
+        s3.putObject(params, function (err, data) {
+            return callback(err, data);
+        });
+    } else {
+        return callback(`AWSC0108 s3 bucketName can not be null or undefined`);
```
Comment (kkuzmin) [inline al_aws.js:235]: please update last message ID comment [here](https://github.com/alertlogic/al-aws-collector-js/pull/87/files#diff-ce5a6d6f6540462ee76a5a98644dbfd0fbc9097588ef11ba6db22e092a270eb4R7)

File: al_aws.js [hunk 3]
```diff
@@ -216,6 +216,25 @@ var setEnv = function(vars, callback) {
...
+    let bucket = bucketName ? bucketName : process.env.aws_lambda_s3_bucket;
+    if (bucket) {
+        const params = {
+            Bucket: bucket,
+            Key: key,
+            Body: parseData
+        };
+        // Uploading files to the bucket
+        s3.putObject(params, function (err, data) {
+            return callback(err, data);
```
Comment (kkuzmin) [inline al_aws.js:232]: I think in this case you can just use
```
s3.putObject(params, callback);
```

**Review by kkuzmin — CHANGES_REQUESTED** [review]

File: al_aws.js [hunk 4]
```diff
@@ -216,6 +216,23 @@ var setEnv = function(vars, callback) {
     });
 };
 
+var uploadS3Object = function ({ data, key, bucketName }, callback) {
+    var s3 = new AWS.S3();
+    // Setting up S3 upload parameters
+    const parseData = typeof data !== 'string' ? JSON.stringify(data) : data;
+    let bucket = bucketName ? bucketName : process.env.aws_lambda_s3_bucket;
```
Comment (kkuzmin) [inline al_aws.js:223]: we should be using different S3 bucket created for DLQ data specifically.

**Review by kkuzmin — CHANGES_REQUESTED** [review]

File: al_aws.js (hunk 4 above)
Comment (rakhimundhada15) [reply al_aws.js]: created new bucket and updating as  dL_s3_bucket_name

File: al_aws.js [hunk 5]
```diff
@@ -216,6 +216,23 @@ var setEnv = function(vars, callback) {
     });
 };
 
+var uploadS3Object = function ({ data, key, bucketName }, callback) {
+    var s3 = new AWS.S3();
+    // Setting up S3 putObject parameters
+    const parseData = typeof data !== 'string' ? JSON.stringify(data) : data;
+    let bucket = bucketName ? bucketName : process.env.dL_s3_bucket_name;
```
Comment (kkuzmin) [inline al_aws.js:223]: let's use lower case `dL_s3_bucket_name `

File: health_checks.js [hunk 6]
```diff
@@ -121,10 +124,25 @@ function formatError(code, exception, type) {
...
 
+/**
+ * 
+ * @param {*} error 
+ * @param {*} param1 - S3 putObject parameters
+ * @param {*} callback 
+ * @returns 
+ */
+function handleIngestEncodingInvalidError(err, { data, key, bucketName }, callback) {
+    if (err.httpErrorCode === INGEST_INVALID_ENCODING.code && err.message.includes(INGEST_INVALID_ENCODING.message)) {
```
Comment (kkuzmin) [inline health_checks.js:135]: I suggest we use error code only, because error message can be changed in future.

**Review by kkuzmin — CHANGES_REQUESTED** [review]

File: health_checks.js (hunk 6 above)
Comment (rakhimundhada15) [reply health_checks.js:135]: I will remove

**Review by kkuzmin — APPROVED** [review]

**Review by kkuzmin — APPROVED** [review]

**Pull request merged**